src/rdo.rs

```diff
@@ -240,41 +242,51 @@ pub fn rdo_mode_decision(
     };
 
     // Find the best chroma prediction mode for the current luma prediction mode
-    let cfl = &CFLParams::new();
     for &chroma_mode in &mode_set_chroma {
+      let mut cfl_alpha_set = vec![ [-1, 0], [-1, 1], [-1, -1], [1, -1], [0, -1], [1, 0], [0, 1], [1, 1] ];
+      if chroma_mode != PredictionMode::UV_CFL_PRED {
+        cfl_alpha_set.truncate(1);
+      } else if !best_mode_chroma.is_intra() {
+        continue;
+      }
       for &skip in &[false, true] {
         // Don't skip when using intra modes
         if skip && luma_mode.is_intra() { continue; }
 
-        let mut wr: &mut dyn Writer = &mut WriterCounter::new();
-        let tell = wr.tell_frac();
-
-
-        encode_block_a(seq, cw, wr, bsize, bo, skip);
-        encode_block_b(fi, fs, cw, wr, luma_mode, chroma_mode, ref_frame, mv, bsize, bo, skip, seq.bit_depth, cfl);
-
-        let cost = wr.tell_frac() - tell;
-        let rd = compute_rd_cost(
-          fi,
-          fs,
-          w,
-          h,
-          is_chroma_block,
-          bo,
-          cost,
-          seq.bit_depth
-        );
-
-        if rd < best_rd {
-          best_rd = rd;
-          best_mode_luma = luma_mode;
-          best_mode_chroma = chroma_mode;
-          best_ref_frame = ref_frame;
-          best_mv = mv;
-          best_skip = skip;
+        for &[alpha_u, alpha_v] in &cfl_alpha_set {
```

Comment:
I think this alpha parameter search sholud be done only when chroma_mode == CFL_PRED?
Also, you might want to add speed condition like, "&& fi.config.speed <= "?

Reply:
I also see that UV_CFL_PRED is neither in RAV1E_INTRA_MODES_MINIMAL or RAV1E_INTRA_MODES, did you mean this?

Reply:
Also, is there any specific reason why the alpha search loop is inside skip test loop?
Because, we would move the skip testing into inside encode_block_b(), in order to fix the redundant call of predictions for non-skip and skip cases.